### PR TITLE
Bb cache

### PR DESCRIPTION
**Pull request by PozhiloyPumba**

### CODE REVIEW

**Review by MrYar0s — DISMISSED** [review]

File: simulator/hart_impl.cpp [hunk 1]
```diff
@@ -20,11 +20,19 @@ void Hart::RunImpl(Mode mode)
...
             interpreter::BB raw_bb;
             interpreter::DecodedBB decodedBB;
+            Register cache_addr;
             do {
-                fetch_.loadBB(executor_.getPC(), raw_bb);
-                decoder_.DecodeBB(raw_bb, decodedBB);
-                executor_.RunBB(decodedBB);
-                counter += raw_bb.size();
+                cache_addr = executor_.getPC() / 4 % BB_CACHE_SIZE;
+                if (auto &&el = bb_cache_[cache_addr]; el.first == executor_.getPC()) {
```
Comment (MrYar0s) [inline simulator/hart_impl.cpp:26]: I think that this change will make it more readable
```suggestion
                if (auto &&[addr, decoded_bb]= bb_cache_[cache_addr]; addr == executor_.getPC()) {
```

Comment (MrYar0s) [reply simulator/hart_impl.cpp:26]: And 2 lines bottom offcourse the same

File: simulator/hart.h [hunk 2]
```diff
@@ -25,6 +27,8 @@ class Hart final {
     interpreter::Fetch fetch_;
     interpreter::Decoder decoder_;
     interpreter::Executor executor_;
+    static inline constexpr const size_t BB_CACHE_SIZE = 256;
```
Comment (MrYar0s) [inline simulator/hart.h:30]: Seems overcomplex for me, @Mirageinvo pls check and give ur opinion

Comment (Mirageinvo) [reply simulator/hart.h:30]: Discussed in chat

**Review by MrYar0s — APPROVED** [review]

**Review by Mirageinvo — APPROVED** [review]

**Pull request merged**